addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/HttpHandlerFactory.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.http.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.link.ItemChannelLinkRegistry;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.openhab.binding.http.handler.HttpThingHandler;
+import org.openhab.binding.http.model.HttpHandlerFactoryConfig;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.openhab.binding.http.HttpBindingConstants.THING_TYPE_HTTP;
+
+/**
+ * Handler factory for creating handlers for things of type http.
+ *
+ * @author Brian J. Tarricone
+ */
+@NonNullByDefault
+@Component(service = ThingHandlerFactory.class, configurationPid = "binding.http")
+public class HttpHandlerFactory extends BaseThingHandlerFactory {
+    private final Map<ThingUID, HttpThingHandler> handlers = new HashMap<>();
+
+    private HttpClient httpClient;
```

Comment:
The static code checker will not like this line. The reason is `@NonNullByDefault`. That annotation enforces that every field is either annotated with `@Nullable` (which in our case is only partly correct, because OSGI will in every case set this for us), or the field is set in the constructor, so that it can not be null.

To tell the code checker that you yourself will care about this not being null, annotate it with `@NonNullByDefault({})` like in:
```
private @NonNullByDefault({}) HttpClient httpClient;
```